### PR TITLE
Feature/aa 173 add incremental mapping

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**

### CODE REVIEW

File: HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs [hunk 1]
```diff
@@ -90,7 +91,8 @@ public IActionResult CancelAccommodationDataCalculation()
         /// <returns></returns>
         [HttpPost("preloading/start")]
         [ProducesResponseType((int) HttpStatusCode.Accepted)]
-        public IActionResult Preload([FromBody] List<Suppliers> suppliers, CancellationToken cancellationToken = default)
+        public IActionResult Preload([FromBody] List<Suppliers> suppliers,
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs:94]: Formatting, please

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs [hunk 2]
```diff
@@ -53,42 +53,43 @@ public async Task Calculate(List<Suppliers> suppliers, CancellationToken cancell
...
+                        if (changedSupplierHotelCodes.Count == 0)
+                            break;
+
+                        var parameters = new List<string>(changedSupplierHotelCodes);
+                        parameters.Add(supplier.ToString().ToLower());
+
+                        //TODO: remove raw sql when ef core will support queries with dictionaries
+                        var notCalculatedAccommodations = await _context.Accommodations
+                            .FromSqlRaw(
+                                @$"SELECT * FROM ""Accommodations"" a
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs:83]: You may get table and column names from an entity. I believe @oleg1std did such things before

File: HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs [hunk 3]
```diff
@@ -169,7 +207,8 @@ public IActionResult MergeAccommodationsData()
                 {
                     try
                     {
-                        var accommodationService = scope.ServiceProvider.GetRequiredService<IAccommodationsDataMerger>();
+                        var accommodationService =
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs:210]: Formatting

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs (hunk 2 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs]: I have changed. Do You mean that?

File: HappyTravel.Nakijin.Api/Models/Mappers/Enums/MappingTypes.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+namespace HappyTravel.Nakijin.Api.Models.Mappers.Enums
+{
+    public enum MappingTypes
+    {
+        Full = 1,
```
Comment (mingazhev) [inline HappyTravel.Nakijin.Api/Models/Mappers/Enums/MappingTypes.cs:5]: Please add "None" with 0 there for more consistency and better error checks if needed

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs [hunk 5]
```diff
@@ -53,42 +53,43 @@ public async Task Calculate(List<Suppliers> suppliers, CancellationToken cancell
...
+                            .Select(ac => ac.SupplierAccommodationId)
+                            .ToListAsync(cancellationToken);
+
+                        if (changedSupplierHotelCodes.Count == 0)
+                            break;
+
+                        var parameters = new List<string>(changedSupplierHotelCodes);
+                        parameters.Add(supplier.ToString().ToLower());
+
+                        //TODO: remove raw sql when ef core will support queries with dictionaries
```
Comment (mingazhev) [inline HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs:80]: Please add a whitespace after "//"

**Review by mingazhev — APPROVED** [review]
With little comments

File: HappyTravel.Nakijin.Api/Models/Mappers/Enums/MappingTypes.cs (hunk 4 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Api/Models/Mappers/Enums/MappingTypes.cs:5]: For me enum must have default value, when we have separate behavior according that value. But if it is our code standard to have everywhere default value, I will add.

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs (hunk 2 above)
Comment (kirillta) [reply HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs]: No, [that one](https://stackoverflow.com/questions/45667126/how-to-get-table-name-of-mapped-entity-in-entity-framework-core)

**Review by kirillta — COMMENTED** [review]
Just waiting for updates

**Review by kirillta — APPROVED** [review]

**Pull request merged**